falcon/response.py

```diff
@@ -152,9 +152,8 @@ def set_header(self, name, value):
                 may be used on platforms that use wide characters.
 
         """
-
         # NOTE(kgriffs): normalize name by lowercasing it
-        self._headers[name.lower()] = value
+        self._headers[str(name).lower()] = str(value)
```

Comment:
Would you mind adding a comment here, e.g.:

`PERF(ginoledesma): Using str() is faster that explicitly checking blah blah`